uaclient/api/u/pro/security/vulnerabilities/_common/v1.py

```diff
@@ -0,0 +1,352 @@
+                affected_pkg
+            ).items():
+                vuln_info = vulns_info.get(vuln_name, "")
+                vuln_fixed_version = vuln.get("source_fixed_version")
+                vuln_status = vuln.get("status")
+
+                # if the vulnerability fixed version is None,
+                # that means that no fix has been published
+                # yet.
+                if vuln_fixed_version is None:
+                    if vuln_status != "not-vulnerable":
+                        if vuln_name not in vulnerabilities:
+                            vulnerabilities[vuln_name] = vuln_info
+                            vulnerabilities[vuln_name][
+                                "affected_packages"
+                            ] = []
+
+                        vulnerabilities[vuln_name]["affected_packages"].extend(
+                            {
+                                "name": pkg_name,
+                                "current_version": pkg_version,
+                                "fix_version": None,
+                                "status": vuln_status,
+                                "fix_available_from": None,
+                            }
+                            for pkg_name, pkg_version in sorted(
+                                binary_pkgs.items()
+                            )
+                        )
+
+                    continue
+
+                for pkg_name, pkg_version in sorted(binary_pkgs.items()):
+                    try:
+                        pocket = source_version[vuln_fixed_version].get(
+                            "pocket"
+                        )
+                        fix_version = (
+                            source_version[vuln_fixed_version]
+                            .get("binary_packages", {})
+                            .get(pkg_name, "")
+                        )
+                    except KeyError:
+                        # There is bug in the data where some sources are
+                        # not present. The Security team is already aware
+                        # of this issue and they are handling it
+                        continue
+
+                    if apt.version_compare(fix_version, pkg_version) > 0:
+                        if vuln_name not in vulnerabilities:
+                            vulnerabilities[vuln_name] = vuln_info
+                            vulnerabilities[vuln_name][
+                                "affected_packages"
+                            ] = []
+
+                        vulnerabilities[vuln_name]["affected_packages"].append(
+                            {
+                                "name": pkg_name,
+                                "current_version": pkg_version,
+                                "fix_version": fix_version,
+                                "status": vuln_status,
+                                "fix_available_from": pocket,
+                            }
+                        )
```

Comment:
@panlinux @basak @dodys please take a look on the logic where we say if a CVE affects the system or not. I just want to double check if I am not missing anything